sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -4,22 +4,30 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Azure.Core.Pipeline
 {
     /// <summary>
     /// A policy that sends an <see cref="AccessToken"/> provided by a <see cref="TokenCredential"/> as an Authentication header.
     /// </summary>
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
```

Comment:
`SemaphoreSlim` is required to be disposed, need to check how expensive it is to recreate new instance every time it is needed.

Reply:
Fixed